CHANGES.md

```diff
@@ -1,9 +1,519 @@
+
+Compilation Time Improvements
+-----------------------------
+
+Generated Code Improvements
+---------------------------
+* added `const` qualifiers for `c_ptrConst` types in generated C code
+
+Memory Improvements
+-------------------
+
+Syntax Highlighting
+-------------------
+
+Configuration / Build Changes
+-----------------------------
+* added support for building with `CHPL_HWLOC=system` and/or `CHPL_GMP=system`  
+  (see https://chapel-lang.org/docs/2.1/usingchapel/chplenv.html#chpl-hwloc  
+   and https://chapel-lang.org/docs/2.1/usingchapel/chplenv.html#chpl-gmp)
+* added an error message if the user attempts to set `CHPL_RE2=system`
+* improved error messages when the Clang version doesn't match LLVM
+* added `chpl-shim` to `make install` to make it available in the user's path
+* added `printchplenv --ignore-errors` to continue past errors encountered
+* removed some unnecessary `__pycache__` files from the released source tarball
+
+Portability / Platform-specific Improvements
+--------------------------------------------
+* improved portability for various compilers and libraries  
+  (e.g. LLVM, GCC, Clang, CUDA, and OFI)
+* removed usage of non-standard C++17 variable length arrays
+* fixed memory leaks in the runtime when using `CHPL_COMM=ofi` with AWS EFA
+* fixed regex escape sequences generating errors with Python 3.12
+* fixed some bugs with `CHPL_COMM=ugni` when using bundled LLVM and/or GPUs
+* fixed a portability issue  in which the PGI compiler's name was mis-spelled
+
+Compiler Improvements
+---------------------
+* updated `chpl` to support LLVM 18
+* improved information generated by `--explain-call` for return intent overloads
+
+Compiler Flags
+--------------
+* removed the deprecated `--llvm` flag in favor of `--target-compiler=llvm`
+
+Generated Executable Flags
+--------------------------
+* added a new `--gpus-per-node` flag for SLURM-based launchers
+
+Error Messages / Semantic Checks
+--------------------------------
+* improved errors when modifying `const` shadow variables in nested `forall`s
+* added an error message for calling `export proc`s with wrong number of args
+* improved error message for improper uses of `_`, the 'throwaway' variable
+* added an error when using `var` intents with `foreach` loops
+* added warnings for copy-initing maps/sets/lists with mismatched `parSafe`s  
+  (see https://chapel-lang.org/docs/2.1/modules/standard/Map.html#Map.warnForMapParsafeMismatch,  
+       https://chapel-lang.org/docs/2.1/modules/standard/Set.html#Set.warnForSetParsafeMismatch,  
+   and https://chapel-lang.org/docs/2.1/modules/standard/List.html#List.warnForListParsafeMismatch)
+
+Launchers
+---------
+* updated the `pbs-gasnetrun_ibv` launcher to request exclusive access to nodes
+
+Runtime Library Changes
+-----------------------
+* updated the runtime to use the GASNet-EX API by default for `CHPL_COMM=gasnet`
+* updated GASNet to use the same copy of `hwloc` as Chapel does
```

Comment:
Suggest rewording this slightly, to reflect the fact that previously Chapel was force-disabling GASNet's hwloc support

```suggestion
* enable GASNet use of `hwloc`, with the same copy as Chapel
```

Reply:
Thanks for clarifying.  I took a slightly different wording, but hopefully captured the change accurately:

```
* enabled GASNet's `hwloc` support, using the same copy of `hwloc` as Chapel
```